testing/installtest/checks_unix.go

```diff
@@ -0,0 +1,149 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !windows
+
+package installtest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/install"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+)
+
+func checkPlatform(_ *atesting.Fixture, topPath string, unprivileged bool) error {
+	if unprivileged {
+		// Check that the elastic-agent user/group exist.
+		uid, err := install.FindUID(install.ElasticUsername)
+		if err != nil {
+			return fmt.Errorf("failed to find %s user: %w", install.ElasticUsername, err)
+		}
+		gid, err := install.FindGID(install.ElasticGroupName)
+		if err != nil {
+			return fmt.Errorf("failed to find %s group: %w", install.ElasticGroupName, err)
+		}
+
+		// Ensure entire installation tree has the correct permissions.
+		err = validateFileTree(topPath, uint32(uid), uint32(gid))
+		if err != nil {
+			// context already added
+			return err
+		}
+
+		// Check that the socket is created with the correct permissions.
+		socketPath := filepath.Join(topPath, paths.ControlSocketName)
+		err = waitForNoError(context.Background(), func(_ context.Context) error {
```

Comment:
You should accept a context as an argument through `CheckSuccess` so that this cancels or expires when the context in the test calling it does.

Reply:
Fixed.